mlir/lib/Dialect/ArmSME/Transforms/OuterProductFusion.cpp

```diff
@@ -31,6 +31,65 @@ using namespace mlir;
 using namespace mlir::arm_sme;
 
 namespace {
+
+// Common match failure reasons.
+static constexpr StringLiteral
+    matchFailureNoAccumulator("no accumulator operand");
+static constexpr StringLiteral matchFailureExpectedOuterProductDefOp(
+    "defining op of accumulator must be 'arm_sme.outerproduct'");
+static constexpr StringLiteral matchFailureInconsistentCombiningKind(
+    "combining kind (add or sub) of outer products must match");
+static constexpr StringLiteral matchFailureInconsistentMasking(
+    "unsupported masking, either both outerproducts are masked "
+    "or neither");
+static constexpr StringLiteral matchFailureOuterProductNotSingleUse(
+    "outer product(s) not single use and cannot be removed, no benefit to "
```

Comment:
very nit: It seems more common to prefix constants with `k`, so `kMatch...`

Reply:
+1